server/text_generation_server/models/causal_lm.py

```diff
@@ -376,40 +378,50 @@ def from_pb(
         tokenized_inputs = tokenizer(
             [r.data.inputs for r in requests] + dummy_inputs,
             return_tensors="pt",
-            padding="max_length",
+            padding="longest",
             return_token_type_ids=False,
             truncation=True,
             max_length=max_input_length,
         )
 
         input_len = tokenized_inputs["input_ids"].shape[1]
+
+        bucket_size = max_input_length
+        left_padding = max_input_length - input_len
+        if input_len < max_input_length and PAD_SEQUENCE_TO_MULTIPLE_OF != 0:
+            assert PAD_SEQUENCE_TO_MULTIPLE_OF <= max_input_length, "PAD_SEQUENCE_TO_MULTIPLE_OF cannot be higher than max_input_length"
+            buckets = list(range(PAD_SEQUENCE_TO_MULTIPLE_OF, max_input_length + 1, PAD_SEQUENCE_TO_MULTIPLE_OF))
+            bucket_idx = bisect.bisect(buckets, input_len)
+            bucket_size = buckets[bucket_idx] - 1
```

Comment:
you can use `round_up` function to do that (it's defined in the same file).

`bucket_size = round_up(max_input_length + 1, PAD_SEQUENCE_TO_MULTIPLE_OF)`